coders/dds.c

```diff
@@ -1624,6 +1832,355 @@ static MagickBooleanType ReadDXT5(const ImageInfo *image_info,Image *image,
     return(SkipDXTMipmaps(image,dds_info,16,exception));
 }
 
+static unsigned char GetBit(const unsigned char *block,size_t *startBit)
+{
+  size_t index=(*startBit) >> 3;
+  size_t base=(*startBit) - (index << 3);
+  (*startBit)++;
+  return ((block[index] >> base) & 0x01);
+}
+
+static unsigned char GetBits(const unsigned char *block,size_t *startBit,
+  unsigned char numBits)
+{
+  size_t index=(*startBit) >> 3;
+  size_t base=(*startBit) - (index << 3);
+  unsigned char ret=0;
+  if (base + numBits > 8)
+  {
+    size_t uFirstIndexBits=8 - base;
+    size_t uNextIndexBits=numBits - uFirstIndexBits;
+    ret=((block[index] >> base) | (((block[index + 1]) &
+      ((1u << uNextIndexBits) - 1)) << uFirstIndexBits));
+  }
+  else
+  {
+    ret=((block[index] >> base) & ((1 << numBits) - 1));
+  }
+  (*startBit)+=numBits;
+  return ret;
+}
+
+static MagickBooleanType IsPixelAnchorIndex(unsigned char subsetIndex,
+  unsigned char numSubsets,size_t pixelIndex,unsigned char partitionid)
+{
+  /* for first subset */
+  if (subsetIndex == 0)
+    return AnchorIndexTable[0][partitionid] == pixelIndex;
+  /* for second subset of two subset partitioning */
+  if (subsetIndex == 1 && numSubsets == 2)
+    return AnchorIndexTable[1][partitionid] == pixelIndex;
+  /* for second subset of three subset partitioning */
+  if (subsetIndex == 1 && numSubsets == 3)
+    return AnchorIndexTable[2][partitionid] == pixelIndex;
+
+  /* for third subset of three subset partitioning */
+  return AnchorIndexTable[3][partitionid] == pixelIndex;
+}
+
+static MagickBooleanType ReadEndpoints(BC7Colors *endpoints,
```

Comment:
The return value is always `MagickTrue` and not check. Can we change this to `void` instead?